packages/api/lambdas/sf-starter.js

```diff
@@ -27,33 +42,143 @@ function dispatch(message) {
 }
 
 /**
- * This is an SQS Queue consumer.
+ * Increment the priority semaphore.
  *
- * It reads messages from a given sqs queue based on the configuration provided
- * in the event object
+ * @param {string} key - Key for the priority semaphore
+ * @param {number} maximum - Maximum number of executions allowed for this semaphore
+ * @returns {Promise}
+ * @throws {Error}
+ */
+async function incrementPrioritySemaphore(key, maximum) {
+  const semaphore = new Semaphore(
+    dynamodbDocClient(),
+    process.env.SemaphoresTable
+  );
+
+  try {
+    await semaphore.up(key, maximum);
+  } catch (err) {
+    if (err instanceof ResourcesLockedError) {
+      log.info(`Unable to start new execution: the maximum number of executions for ${key} are already running.`);
+    }
+    throw err;
+  }
+}
+
+/**
+ * Attempt to increment the priority semaphore and start a new execution.
+ *
+ * If `incrementPrioritySemaphore()` is unable to increment the priority semaphore,
+ * it throws an error and `dispatch()` is not called.
+ *
+ * @param {Object} queueMessage - SQS message
+ * @returns {Promise} - Promise returned by `dispatch()`
+ * @throws {Error}
+ */
+async function incrementAndDispatch(queueMessage) {
+  const message = get(queueMessage, 'Body');
```

Comment:
Not a hill I'm going to die on, but `get()` is not necessary here.

```suggestion
  const message = queueMessage.Body);
```